### PR TITLE
the big nazzo

### PR DESCRIPTION
**Pull request by GabeLoins**
@vqtran now we can display the audio output of Alexa to the user.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+gulp.task('run', ['package'], function() {
+  if (server) {
+    server.kill('SIGINT');
+  }
+
+  var target = child.execSync('ls target/echoquery-1.1-jar-with-dependencies.jar') + '';
+  var deps =
+      child.execSync('mvn dependency:build-classpath | grep -v INFO') + '';
+  server = child.spawn('java', [
+      '-cp', target.trim() + ':' + deps.trim(), 'echoquery.frontend.Main', '"$*"'
```
Comment (GabeLoins) [inline gulpfile.js]: I made the frontend inside the echoquery package, not 100% I like this decision

File: src/main/java/echoquery/frontend/Main.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  }
+
+  private static void runSparkServer() {
+    Spark.externalStaticFileLocation("public/");
+
+    Spark.get("/",
+        (request, response) -> new ModelAndView(new HashMap<>(), "index.jade"),
+        new JadeTemplateEngine());
+
+    Spark.get("/test", (request, response) -> "Hello World!");
```
Comment (GabeLoins) [inline src/main/java/echoquery/frontend/Main.java]: not really needed but probably good to keep around for debugging

File: src/main/java/echoquery/frontend/Main.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    
+    Spark.get("/fetch", (request, response) ->
```
Comment (GabeLoins) [inline src/main/java/echoquery/frontend/Main.java]: /fetch gets the updated display text

File: src/main/java/echoquery/utils/VisualizationUtil.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+public class VisualizationUtil {
+  private static final Logger log =
+      LoggerFactory.getLogger(VisualizationUtil.class);
+  private static Connection conn = SingletonConnections.getStateInstance();
+
+  public static void updateDisplayText(String message) {
+    try {
+      Statement statement = conn.createStatement();
+      statement.executeUpdate(
+          "update sessions set display='" + StringEscapeUtils.escapeJava(message) + "'where id=1");
```
Comment (GabeLoins) [inline src/main/java/echoquery/utils/VisualizationUtil.java]: right now it only works for one session and doesn't reset between sessions, but as a future PR I'll actually use session id here

File: src/main/resources/js/components/App.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        {this.state.displayText}
+      </div>
+    );
+  }
+
+  componentDidMount() {
+    SessionStore.listen(this.onChange);
+    this.timer = setInterval(() => {
+      SessionUtils.giveSessionDisplayText(DisplayActions.setDisplayText);
+    }, 50);
```
Comment (GabeLoins) [inline src/main/resources/js/components/App.js]: update the screen every 50 milliseconds

**Comment by vqtran** [comment]
Lets .gitignore the bundle.js file so then we don't blow up the line count. I'll take a closer look in two days haha I'm kind of on vaca mode

**Comment by vqtran** [comment]
this looks fine to me, good to merge!

**Comment by vqtran** [comment]
just ran this, it's pretty cool!

**Pull request merged**